### PR TITLE
Fix clippy warnings on Windows

### PR DESCRIPTION
**Pull request by ChrisDenton**
This is likely best reviewed commit-by-commit. The first commit simply uses `cargo clippy --fix` to automatically fix warnings so there almost certainly aren't any behavioural changes. The other commits are manual fixes but should be fine too. The most tricky one is probably 397dd22 because it's rearranging boolean expressions.

### CODE REVIEW

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -1956,8 +1947,8 @@ impl Build {
         // Target flags
         match cmd.family {
             ToolFamily::Clang { .. } => {
-                if !cmd.has_internal_target_arg
-                    && !(target.contains("android")
+                if !(cmd.has_internal_target_arg
+                    || target.contains("android")
                         && android_clang_compiler_uses_target_arg_internally(&cmd.path))
```
Comment (NobodyXu) [inline src/lib.rs:1952]: Same here, I think we need to call the ()

File: src/lib.rs [hunk 2]
```diff
@@ -3370,7 +3353,7 @@ impl Build {
         // CROSS_COMPILE is of the form: "arm-linux-gnueabi-"
         let cc_env = self.getenv("CROSS_COMPILE");
         let cross_compile = cc_env.as_ref().map(|s| s.trim_end_matches('-').to_owned());
-        cross_compile.or(linker_prefix).or(match &target[..] {
+        cross_compile.or(linker_prefix).or(match target {
```
Comment (NobodyXu) [inline src/lib.rs:3356]: IMO we'd need

```suggestion
        cross_compile.or(linker_prefix).or_else(|| match target {
```

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (ChrisDenton) [reply src/lib.rs]: Done. Changing it does affect the formatting.

**Review by NobodyXu — APPROVED** [review]
Thanks!

**Pull request merged**